src/Engine/ProtoCore/AssociativeGraph.cs

```diff
@@ -16,6 +16,291 @@ public enum UpdateStatus
     public class Utils
     {
         /// <summary>
+        /// Builds the dependencies within the list of graphNodes
+        /// </summary>
+        /// <param name="graphNodeScopeToCheck"></param>
+        public static void BuildGraphNodeDependencies(List<AssociativeGraph.GraphNode> graphNodesInScope)
+        {
+            if (graphNodesInScope == null)
```

Comment:
Right now, i cant throw an exception. There are cases where a block of code will be empty. And instead of filtering them out at the caller, they are filtered here